### PR TITLE
added map-leaflet component and enable flow component

### PR DESCRIPTION
**Pull request by Rahul139-stack**
This PR includes:

Implemented Leaflet map component with Marker
Authorizable component and datasource json and tms connected.
**Sample**:
image
Please and approve it

### CODE REVIEW

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/dialog/.content.json [hunk 1]
```diff
@@ -1,15 +1,103 @@
...
+          "label": "Marker Latitude",
+          "description": "These will be added as marker latitude"
+        },
+        "marker-long": {
+          "sling:resourceType": "wcm/dialogs/components/textfield",
+          "name": "markerLng",
+          "label": "Marker longitude",
+          "description": "These will be added as marker longitude"
+        },
+        "popup-text": {
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/map/dialog/.content.json:68]: Please follow the camel casing standard.

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/dialog/.content.json [hunk 2]
```diff
@@ -1,15 +1,103 @@
...
+        "name": "markerList",
+        "label": "Marker Details",
+        "sling:resourceType": "wcm/dialogs/components/multifield",
+        "marker-lat": {
+          "sling:resourceType": "wcm/dialogs/components/textfield",
+          "name": "markerLat",
+          "label": "Marker Latitude",
+          "description": "These will be added as marker latitude"
+        },
+        "marker-long": {
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/map/dialog/.content.json:62]: Please follow the camel casing standard.

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/dialog/.content.json [hunk 3]
```diff
@@ -1,15 +1,103 @@
...
+      }
+    },
+    "markerTab": {
+      "sling:resourceType": "wcm/dialogs/components/tab",
+      "label": "Marker",
+      "markerList": {
+        "name": "markerList",
+        "label": "Marker Details",
+        "sling:resourceType": "wcm/dialogs/components/multifield",
+        "marker-lat": {
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/map/dialog/.content.json:56]: Please follow the camel casing standard.

In most cases, the key and name attributes should be the same.

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/templates/flowsample.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    "mapLat":"46.76336",
+    "mapLng":"-71.32453",
+    "markers":[{
+        "markerLat":"46.76186",
+        "markerLng":"-71.32247",
+        "popupText":"Marker 1"
+    },{
+        "markerLat":"46.76672",
+        "markerLng":"-71.32919",
+        "popupText":"Marker 1"
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/map/templates/flowsample.json:11]: Please provide an update to the popup text for Marker 2.

Even though this is sample data, make sure it looks good.

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/variant.html [hunk 5]
```diff
@@ -1,22 +1,10 @@
...
-    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-    See the License for the specific language governing permissions and
-    limitations under the License.
-*/-->
 <template data-sly-template.variant="${ @ model }">
-  <div 
+  <div
+    id="${model.id}"
     component="${model.componentName}"
+    data-path="${model.componentPath}"
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/map/variant.html:5]: Are we using this data-path anywhere ?

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/clientlibs/behaviour.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+window.Typerefinery = window.Typerefinery || {};
+window.Typerefinery.Components = Typerefinery.Components || {};
+window.Typerefinery.Components.Widgets = Typerefinery.Components.Widgets || {};
+window.Typerefinery.Components.Widgets.Map = Typerefinery.Components.Widgets.Map || {};
+window.Typerefinery.Components.Widgets.Map.LeafletMap = Typerefinery.Components.Widgets.Map.LeafletMap || {};
+console.log("behaviour.js")
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/map/clientlibs/behaviour.js:6]: Please remove the console which is added for debugging purposes.

**Review by Harikarthyk — CHANGES_REQUESTED** [review]

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/dialog/.content.json (hunk 1 above)
Comment (Rahul139-stack) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/map/dialog/.content.json]: Done

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/dialog/.content.json (hunk 2 above)
Comment (Rahul139-stack) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/map/dialog/.content.json]: Done

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/templates/flowsample.json (hunk 4 above)
Comment (Rahul139-stack) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/map/templates/flowsample.json]: Done

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/variant.html (hunk 5 above)
Comment (Rahul139-stack) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/map/variant.html]: No,it is now removed

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/clientlibs/behaviour.js (hunk 6 above)
Comment (Rahul139-stack) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/map/clientlibs/behaviour.js]: Done

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/clientlibs/.content.json [hunk 7]
```diff
@@ -1,16 +1,18 @@
...
   "css": [
-    "/apps/typerefinery/clientlibs/vendor/maplibre/maplibre-gl.css",
+    "/apps/typerefinery/clientlibs/vendor/map/leafletmap/leaflet.css",
     "style.css"
   ],  
   "js": [
-    "/apps/typerefinery/clientlibs/vendor/maplibre/maplibre-gl.js",
-    "script.js"
+    "/apps/typerefinery/clientlibs/vendor/map/leafletmap/leaflet.js",
+    "mock/coordinates.js",
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/map/clientlibs/.content.json:14]: I think this file is deleted from the src, then why is this included here ?

**Review by Harikarthyk — CHANGES_REQUESTED** [review]

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/map/clientlibs/.content.json (hunk 7 above)
Comment (Rahul139-stack) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/map/clientlibs/.content.json]: Done

**Comment by wildone** [comment]
where are the tests for this?
